scene/resources/animation.cpp

```diff
@@ -5692,9 +5688,7 @@ Variant Animation::interpolate_variant(const Variant &a, const Variant &b, float
 			return Vector2i(int32_t(va.x + (vb.x - va.x) * c), int32_t(va.y + (vb.y - va.y) * c));
 		}
 		case Variant::RECT2: {
-			const Rect2 ra = a.operator Rect2();
-			const Rect2 rb = b.operator Rect2();
-			return Rect2(ra.position.lerp(rb.position, c), ra.size.lerp(rb.size, c));
+			return (a.operator Rect2()).lerp(b.operator Rect2(), c);
 		}
 		case Variant::RECT2I: {
 			const Rect2i ra = a.operator Rect2i();
```

Comment:
I guess this fix would be valid for Rect2i as well. However, since it does not define the lerp in the class in the first place, and the interpolation occurs only in animation, so for now it would be fine to just apply Math::abs() to the calculation of the size in the line 5696 for avoiding error.

Reply:
Tweak applied to the code, thanks @TokageItLab!